src/ucs/sys/socket.c

```diff
@@ -0,0 +1,223 @@
+            }
+
+            if (errno == EISCONN) {
+                status = UCS_OK;
+                goto out;
+            }
+
+            if (errno != EINTR) {
+                ucs_error("connect(fd=%d, dest_addr=%s) failed: %m", fd,
+                          ucs_sockaddr_str(dest_addr, str, UCS_SOCKADDR_STRING_LEN));
+                return UCS_ERR_UNREACHABLE;
+            }
+        }
+    } while ((ret < 0) && (errno == EINTR));
+
+out:
+    ucs_debug("connect(fd=%d, dest_addr=%s): %m", fd,
+              ucs_sockaddr_str(dest_addr, str, UCS_SOCKADDR_STRING_LEN));
+    return status;
+}
+
+ucs_status_t ucs_socket_connect_nb_get_status(int fd)
+{
+    socklen_t conn_status_sz;
+    int ret, conn_status;
+
+    conn_status_sz = sizeof(conn_status);
+
+    ret = getsockopt(fd, SOL_SOCKET, SO_ERROR,
+                     &conn_status, &conn_status_sz);
+    if (ret < 0) {
+        ucs_error("getsockopt(fd=%d) failed to get SOL_SOCKET(SO_ERROR): %m", fd);
+        return UCS_ERR_IO_ERROR;
+    }
+
+    if ((conn_status == EINPROGRESS) || (conn_status == EWOULDBLOCK)) {
+        return UCS_INPROGRESS;
+    }
+
+    if (conn_status != 0) {
+        ucs_error("SOL_SOCKET(SO_ERROR) status on fd %d: %s", fd, strerror(conn_status));
+        return UCS_ERR_UNREACHABLE;
+    }
+
+    return UCS_OK;
+}
+
+ucs_status_t ucs_sockaddr_sizeof(const struct sockaddr *addr, size_t *size_p)
+{
+    ucs_status_t status = UCS_OK;
+
+    switch (addr->sa_family) {
+    case AF_INET:
+        *size_p = sizeof(struct sockaddr_in);
+        break;
+    case AF_INET6:
+        *size_p = sizeof(struct sockaddr_in6);
+        break;
+    default:
+        ucs_error("unknown address family: %d", addr->sa_family);
+        status = UCS_ERR_INVALID_PARAM;
+        break;
+    }
+
+    return status;
+}
+
+ucs_status_t ucs_sockaddr_get_port(const struct sockaddr *addr, unsigned *port_p)
+{
+    ucs_status_t status = UCS_OK;
+
+    switch (addr->sa_family) {
+    case AF_INET:
+        *port_p = ntohs(UCS_SOCKET_INET_PORT(addr));
+        break;
+    case AF_INET6:
+        *port_p = ntohs(UCS_SOCKET_INET6_PORT(addr));
+        break;
+    default:
+        ucs_error("unknown address family: %d", addr->sa_family);
+        status = UCS_ERR_INVALID_PARAM;
+        break;
+    }
+
+    return status;
+}
+
+const void *ucs_sockaddr_get_inet_addr(const struct sockaddr *addr)
+{
+    switch (addr->sa_family) {
+    case AF_INET:
+        return &UCS_SOCKET_INET_ADDR(addr);
+    case AF_INET6:
+        return &UCS_SOCKET_INET6_ADDR(addr);
+    default:
+        ucs_error("unknown address family: %d", addr->sa_family);
+        return NULL;
+    }
+}
+
+const char* ucs_sockaddr_str(const struct sockaddr *sock_addr,
+                             char *str, size_t max_size)
+{
+    unsigned port;
+    size_t str_len;
+
+    if ((sock_addr->sa_family != AF_INET) && (sock_addr->sa_family != AF_INET6)) {
+        ucs_error("unknown address family: %d", sock_addr->sa_family);
```

Comment:
remove ucs_error() from this function, since we already return the error as a string

Reply:
removed